### PR TITLE
Add new example plugins contributed by EclipseSource

### PR DESCRIPTION
**Pull request by mathansen**
Add new example plugins contributed by EclipseSource

### CODE REVIEW

File: samples/sample-plugin-embedjs/che-sample-plugin-embedjs-ide/pom.xml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-sample-plugin-embedjs-parent</artifactId>
+        <groupId>org.eclipse.che.sample</groupId>
+        <version>4.5.1</version>
```
Comment (skabashnyuk) [inline samples/sample-plugin-embedjs/che-sample-plugin-embedjs-ide/pom.xml]: <version>4.7.0-SNAPSHOT</version>

File: samples/sample-plugin-embedjs/che-sample-plugin-embedjs-ide/pom.xml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                        <id>analyze</id>
+                        <configuration>
+                            <skip>true</skip>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (skabashnyuk) [inline samples/sample-plugin-embedjs/che-sample-plugin-embedjs-ide/pom.xml]: what for?

File: samples/sample-plugin-embedjs/che-sample-plugin-embedjs-ide/src/main/resources/org/eclipse/che/plugin/embedjsexample/public/helloworld.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+function HelloWorld(element, contents) {
+    element.textContent = contents;
+};
```
Comment (skabashnyuk) [inline samples/sample-plugin-embedjs/che-sample-plugin-embedjs-ide/src/main/resources/org/eclipse/che/plugin/embedjsexample/public/helloworld.js]: new line

File: samples/sample-plugin-embedjs/pom.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-sample-parent</artifactId>
+        <groupId>org.eclipse.che.sample</groupId>
+        <version>4.5.1</version>
```
Comment (skabashnyuk) [inline samples/sample-plugin-embedjs/pom.xml]: <version>4.7.0-SNAPSHOT</version>

File: samples/sample-plugin-nativeaccess/che-sample-plugin-nativeaccess-ide/pom.xml [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-sample-plugin-nativeaccess-parent</artifactId>
+        <groupId>org.eclipse.che.sample</groupId>
+        <version>4.5.1</version>
```
Comment (skabashnyuk) [inline samples/sample-plugin-nativeaccess/che-sample-plugin-nativeaccess-ide/pom.xml]: <version>4.7.0-SNAPSHOT</version>

File: samples/sample-plugin-nativeaccess/che-sample-plugin-nativeaccess-ide/pom.xml [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+                    <execution>
+                        <id>analyze</id>
+                        <configuration>
+                            <skip>true</skip>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (skabashnyuk) [inline samples/sample-plugin-nativeaccess/che-sample-plugin-nativeaccess-ide/pom.xml]: what for?

File: samples/sample-plugin-nativeaccess/pom.xml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-sample-parent</artifactId>
+        <groupId>org.eclipse.che.sample</groupId>
+        <version>4.5.1</version>
```
Comment (skabashnyuk) [inline samples/sample-plugin-nativeaccess/pom.xml]: <version>4.7.0-SNAPSHOT</version>

File: samples/sample-plugin-parts/che-sample-plugin-parts-ide/pom.xml [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-sample-plugin-parts-parent</artifactId>
+        <groupId>org.eclipse.che.sample</groupId>
+        <version>4.4.0-RC1-SNAPSHOT</version>
```
Comment (skabashnyuk) [inline samples/sample-plugin-parts/che-sample-plugin-parts-ide/pom.xml]: <version>4.7.0-SNAPSHOT</version>

File: samples/sample-plugin-parts/che-sample-plugin-parts-ide/pom.xml [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+                        <id>analyze</id>
+                        <configuration>
+                            <skip>true</skip>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (skabashnyuk) [inline samples/sample-plugin-parts/che-sample-plugin-parts-ide/pom.xml]: what for?

File: samples/sample-plugin-parts/pom.xml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-sample-parent</artifactId>
+        <groupId>org.eclipse.che.sample</groupId>
+        <version>4.4.0-RC1-SNAPSHOT</version>
```
Comment (skabashnyuk) [inline samples/sample-plugin-parts/pom.xml]: <version>4.7.0-SNAPSHOT</version>

File: samples/sample-plugin-serverservice/che-sample-plugin-serverservice-ide/pom.xml [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-sample-plugin-serverservice-parent</artifactId>
+        <groupId>org.eclipse.che.sample</groupId>
+        <version>4.4.0-RC1-SNAPSHOT</version>
```
Comment (skabashnyuk) [inline samples/sample-plugin-serverservice/che-sample-plugin-serverservice-ide/pom.xml]: <version>4.7.0-SNAPSHOT</version>

File: samples/sample-plugin-serverservice/che-sample-plugin-serverservice-ide/pom.xml [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+                        <id>analyze</id>
+                        <configuration>
+                            <skip>true</skip>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (skabashnyuk) [inline samples/sample-plugin-serverservice/che-sample-plugin-serverservice-ide/pom.xml]: what for?

File: samples/sample-plugin-serverservice/che-sample-plugin-serverservice-server/pom.xml [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-sample-plugin-serverservice-parent</artifactId>
+        <groupId>org.eclipse.che.sample</groupId>
+        <version>4.4.0-RC1-SNAPSHOT</version>
```
Comment (skabashnyuk) [inline samples/sample-plugin-serverservice/che-sample-plugin-serverservice-server/pom.xml]: <version>4.7.0-SNAPSHOT</version>

File: samples/sample-plugin-serverservice/pom.xml [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-sample-parent</artifactId>
+        <groupId>org.eclipse.che.sample</groupId>
+        <version>4.4.0-RC1-SNAPSHOT</version>
```
Comment (skabashnyuk) [inline samples/sample-plugin-serverservice/pom.xml]: <version>4.7.0-SNAPSHOT</version>

**Comment by mathansen** [comment]
Thank you for your comments! I fixed the mentioned issues.

File: samples/sample-plugin-embedjs/che-sample-plugin-embedjs-ide/src/main/resources/org/eclipse/che/plugin/embedjsexample/public/helloworld.js [hunk 15]
```diff
@@ -0,0 +1,4 @@
+function HelloWorld(
+    element, contents) {
+    element.textContent = contents;
+};
```
Comment (skabashnyuk) [inline samples/sample-plugin-embedjs/che-sample-plugin-embedjs-ide/src/main/resources/org/eclipse/che/plugin/embedjsexample/public/helloworld.js]: new line

Comment (mathansen) [reply samples/sample-plugin-embedjs/che-sample-plugin-embedjs-ide/src/main/resources/org/eclipse/che/plugin/embedjsexample/public/helloworld.js]: fixed

**Comment by vparfonov** [comment]
move code here https://github.com/eclipse/che/pull/2246

**Pull request closed without merging**